library/tests/src/com/google/maps/android/PolyUtilTest.java

```diff
@@ -125,6 +135,28 @@ public void testOnEdge() {
                    makeList(79, big, 90-small, -90, 90, -135));
     }
 
+    public void testLocationIndex() {
+        final double small = 5e-7;  // About 5cm on equator, half the default tolerance.
+        final double big   = 2e-6;  // About 10cm on equator, double the default tolerance.
+
+        // Empty.
+        locationIndexCase(makeList(), new LatLng(0, 0), -1);
+
+        // One point.
+        locationIndexCase(makeList(1, 2), new LatLng(1, 2), 0);
+        locationIndexCase(makeList(1, 2), new LatLng(3, 5), -1);
+
+        // Two points.
+        locationIndexCase(makeList(1, 2, 3, 5), new LatLng(1, 2), 0);
+        locationIndexCase(makeList(1, 2, 3, 5), new LatLng(3, 5), 1);
```

Comment:
Probably here we should have here a return of 0, because (3,5) is considered part of the first segment [(1,2), (3,5)]. The 1 you get is a side-effect of the current implem, which considers [(1,2), (1,2)] as segment0 and [(1,2), (3,5)] as segment1.